compiler-plugin/src/main/java/io/ballerina/lib/data/jsondata/compiler/JsondataTypeValidator.java

```diff
@@ -133,6 +135,14 @@ private boolean isParseFunctionOfStringSource(ExpressionNode expressionNode) {
         if (expressionNode.kind() != SyntaxKind.FUNCTION_CALL) {
             return false;
         }
+        NameReferenceNode nameReferenceNode = ((FunctionCallExpressionNode) expressionNode).functionName();
+        if (nameReferenceNode.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+            return false;
+        }
+        String prefix = ((QualifiedNameReferenceNode) nameReferenceNode).modulePrefix().toString();
```

Comment:
Calling `toString()` in the syntax tree is not recommended. 
```suggestion
        String prefix = ((QualifiedNameReferenceNode) nameReferenceNode).modulePrefix().text();
```